include/maths/CDataFrameAnalysisStateInterface.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#ifndef INCLUDED_ml_maths_CDataFrameAnalysisInterface_h
```

Comment:
As before, I'd prefer a different name than State, but also:
```suggestion
#ifndef INCLUDED_ml_maths_CDataFrameAnalysisStateInterface_h
```

Reply:
Done